ote/src/clj/ote/integration/import/gtfs.clj

```diff
@@ -220,13 +216,63 @@
                       {::t-service/id interface-id})
       nil)))
 
-(defmethod load-transit-interface-url :gtfs [_ db interface-id url last-import-date saved-etag]
-  (load-interface-url db interface-id url last-import-date saved-etag))
+(defmulti validate-interface-zip-package
+          (fn [type byte-array-input] type))
+
+
+(defmethod validate-interface-zip-package :gtfs [_ byte-array-input]
+  (let [file-list (list-zip byte-array-input)]
+
+    (when-not (or
+                (every? file-list ["agency.txt" "stops.txt" "calendar.txt" "trips.txt"])
+                (every? file-list ["agency.txt" "stops.txt" "calendar_dates.txt" "trips.txt"]))
+      (throw (ex-info "Missing required files in gtfs zip file" {:file-names file-list})))))
+
+(defmethod validate-interface-zip-package :kalkati [_ byte-array-input]
+  (let [file-list (list-zip byte-array-input)]
+
+    (when-not (contains? file-list "LVM.xml")
+      (throw (ex-info "Missing required files in kalkati zip file" {:file-names file-list})))))
+
+
+(defn check-interface-zip [type db interface-id url byte-array-data]
+  (try
+    (validate-interface-zip-package type byte-array-data)
+
+    (catch Exception e
+      (log/warn "Error when opening interface zip package from url" url ":" (.getMessage e))
+      (specql/update! db ::t-service/external-interface-description
+                      {::t-service/gtfs-imported (java.sql.Timestamp. (System/currentTimeMillis))
+                       ::t-service/gtfs-import-error (str "Invalid interface package: " (.getMessage e))}
```

Comment:
gtfs-import-error is shown in ui. I think that it should be in finnish or even better, it should be translated.